sql/src/main/java/org/apache/druid/sql/avatica/DruidConnection.java

```diff
@@ -56,9 +56,8 @@
 
   // Typically synchronized by connectionLock, except in one case: the onClose function passed
   // into DruidStatements contained by the map.
-  private final ConcurrentMap<Integer, DruidStatement> statements;
-
   @GuardedBy("connectionLock")
+  private final ConcurrentMap<Integer, DruidStatement> statements;
```

Comment:
There is an explicit exception stated above here, so guarded by is not correct if I'm reading it

Reply:
Hi, @drcrallen . Thanks for your comment. Yep, as the comment says, `statements` should be synchronized by `connectionLock`, so we should put a `GuardedBy` annotation here. And for the explicit exception, we only need to suppress the warning with `SuppressWarnings`.

Reply:
So is the only reason for the `ConcurrentMap` is so that the `remove` call does not need to be synchronized?

Reply:
Yes, the original discussion can be found [here](https://github.com/apache/incubator-druid/pull/6868#discussion_r249639199).